packages/exchange/src/exchange/providers/utils.py

```diff
@@ -10,6 +11,27 @@
 from tenacity import retry_if_exception
 
 
+def get_env_url(key: str, default: str = "") -> httpx.URL:
+    """
+    Returns a valid 'http' or 'https' URL.
+
+    :param key: The environment key
+    :param default: The URL default value
+    :raises ValueError: If the URL scheme is not 'http' or 'https'
+    """
+
+    val = os.environ.get(key, default)
+    if val == "":
+        raise ValueError(f"{key} was empty")
```

Comment:
:nit: should use `KeyError` which is consistent with the non-default value getter `os.environ[key]`. 

```python
[nav] In [1]: import os
         ...:
         ...: os.environ["SHELL"]
         ...: os.environ["SHELL_MISSING"]
         ...:
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[1], line 4
      1 import os
      3 os.environ["SHELL"]
----> 4 os.environ["SHELL_MISSING"]

File <frozen os>:714, in __getitem__(self, key)

KeyError: 'SHELL_MISSING'
```

Reply:
forgot what package i was in – it just occurred to me that we recently wrote up a way to do this on the base provider. could we add the check in there?

https://github.com/block-open-source/goose/blob/06c12b627c4b2da94c05b021a6b2e31539557798/packages/exchange/src/exchange/providers/base.py#L26-L29

Reply:
so you mean add a parameter to `check_env_vars()` like `base_url_key`?

Reply:
exactly! imo we might want to tweak it slightly so it'll dump all the env vars in a single pass rather than one at a time (especially with how much setup is needed for  `AZURE_*`).

not sure if @elenazherdeva is working on that atm, but it was from this https://github.com/block-open-source/goose/pull/116#discussion_r1792744892

Reply:
@lamchau thank you for pinging! I’m not working on it at the moment, but I should fix it. Let me create a follow-up PR based on your comments!